driver-core/src/main/java/com/datastax/driver/core/AbstractSession.java

```diff
@@ -51,6 +51,7 @@ public ResultSet execute(String query, Object... values) {
      */
     @Override
     public ResultSet execute(Statement statement) {
+        checkNotInEventLoop();
```

Comment:
Because the synchronous methods are defined in `AbstractSession`. But now I realize I could also redefine them to do the check and then just call super().